### DISCUSSION TITLE
How to calculate log2 Fold Change from mRNA transcript dataset that has adjusted p values

### DISCUSSION THREAD

**Discussion by John1951** [community/community#81225] [GitHub Education]
### Hi everyone,

Hey, I have a dataset of about 363 mRNA transcripts from 2 treatment groups (triplicates), and FC and adjusted p values seems to be already calculated. I would like to find the log2 fold change in R using edgeR (or anything) and produce a few graphs showing the difference in gene expression between the 2 treatment groups (from the log2 fold change) that would indicate strong candidate genes for subsequent studies. Yes, I have been researching this and there seems to be any number of ways to do this. If someone can help get me going with the most straightforward way, even if it is in Python, I would greatly appreciate it, John

**Comment by faridnec** [comment]
Hey, here's a brief example of the Python approach using `DESeq2` (DESeq2 is a tool for differential gene expression analysis of RNA-seq data):
```bash
import pandas as pd
import deseq2  # Install DESeq2
import matplotlib.pyplot as plt
```
assuming after loading the data by using pandas and assigned it to a variable called `data`, perform normalization and DDS
```bash
dds = deseq2.DESeqDataSetFromMatrix(countData=data.iloc[:, :363], colData=data.iloc[:, 363:], design=design)
dds = dds.DESeq()
results = dds.results()
```
then extract log2 fold change
```bash
log2fc = results["log2FoldChange"]
```
Lastly, you can use Python plotting libraries like `matplotlib` or `seaborn` for `volcano plots` and `heatmaps`.
example of volcano plotting:
```bash
plt.scatter(log2fc, -np.log10(results["pvalue"]), s=4)
plt.axhline(-np.log10(0.05), color="r", linestyle="--")
plt.xlabel("Log2 Fold Change")
plt.ylabel("-Log10(p-value)")
plt.title("Volcano Plot")
plt.show()
```
These are just examples though. I recommend to check some documentations like [this](https://github.com/owkin/PyDESeq2)

**Reply by John1951** [reply]
Hey Faridnec, Thank you so much for your help. I will run the dataset in Python using what you suggested tonight. Ya, thanks, that looks a whole lot easier and less messy than what I was trying to do in R, John

**Reply by faridnec** [reply]
Sure anytime if you run to a problem don't hesitate ask! And I'd like to know the update if it works for you

**Comment by jlchen5** [comment]
Hi John. I have a repo which is made for RNA-seq analysis in R. Maybe you could find some info in there: [https://github.com/jlchen5/CELR-D-22-00015](https://github.com/jlchen5/CELR-D-22-00015) and [RNA-seq-analysis](https://github.com/jlchen5/RNA-seq-Analysis). Let me know if you want to get more details.
